reactor-core/src/main/java/reactor/core/scheduler/ParallelScheduler.java

```diff
@@ -134,13 +137,7 @@ ScheduledExecutorService pick() {
         ScheduledExecutorService[] a = executors;
         if (a != SHUTDOWN) {
             // ignoring the race condition here, its already random who gets which executor
-            int idx = roundRobin;
-            if (idx == n) {
-                idx = 0;
-                roundRobin = 0;
-            } else {
-                roundRobin = idx + 1;
-            }
+            int idx = ROUND_ROBIN.getAndUpdate(this, i -> (i < (n-1)) ? i + 1 : 0);
```

Comment:
Or it's possible to make it 
```
int idx = roundRobin;
            if (idx == n) {
                idx = 0;
                roundRobin = 1;
            } else {
                roundRobin = idx + 1;
            }
```

to keep the correct distribution & make minimum changes.